Submission2

```diff
@@ -0,0 +1,142 @@
+using std::ofstream;
+
+int find_max(const string& infilename)              //DONE this part!
+{
+    int highestnumber, current;
+    std::ifstream infile;
+    infile.open("input.txt");
+    
+    infile >> current;
+    highestnumber = current;
+    
+    while (infile)
+    {
+        infile >> current;
+        if (current > highestnumber)
+            highestnumber = current;
+    }
+    
+    return highestnumber;
+}
+
+void sieve(const int size, bool primes[])          //DONE this part!
+{
+    for (int i = 2; i <= size; i++)
+    {
+        if (primes[i] == true)
+        
+            std::cout << i;
+        
+            for (int j = (i * i); j <= i; j++)
+            {
+                primes[j] = false;
+            }
+    }
+}
+
+int write_primes(const int size, bool primes[], const string& infilename, const string& outfilename) //Incomplete
+{
+    int p;
+    
+    std::ifstream infile("input.txt");
+    if (!infile.is_open())
+        return -1;
+    
+    std::ofstream outfile("primes.txt");
+    if (!outfile.is_open())
+        return -1;
+    
+    while (infile >> p)
+    {
+        infile >> p;
+        if (primes[p] == 1)
+        {
+            outfile << p << endl;
+        }
+    }
+    
+    return p;
+}
+
+int write_composites(const int size, bool primes[], const string& infilename, const string& outfilename) // Incomplete
+{
+    int compositesnumber;
+    
+    std::ifstream infile("input.txt");
+    if (!infile.is_open())
+        return -1;
+    
+    std::ofstream outfile("primes.txt");
+    if (!outfile.is_open())
+        return -1;
+    
+    sieve(size, primes);
+    
+    return compositesnumber;
+}
+
+int main()
+{
+    string line;
+    
+    std::ifstream infilename("input.txt");
+    if (infilename.is_open())
+    {
+        while (getline (infilename, line))
+        {
+            std::cout << line << '\n';
+        }
+    }
+    
+    else return -1;
+    
+    int max = find_max("input.txt");
+    if (max == -1) {
+        std::cout << "ERROR in `find_max()`" << endl;
+        return -1;  // error
+    }
+    
+    bool * primes = (bool *) malloc( sizeof(bool) * (max+1) );
+    sieve(max+1, primes);
+    
+    int ret;  // for storing return values, to check for error codes
+    //
+    ret = write_primes(max+1, primes, "input.txt", "primes.txt");
+    if (ret == -1) {
+        std::cout << "ERROR in `write_primes()`" << endl;
+        return -1;  // error
+    }
+    //
+    ret = write_composites(max+1, primes, "input.txt", "composites.txt");
+    if (ret == -1) {
+        std::cout << "ERROR in `write_composites()`" << endl;
+        return -1;  // error
+    }
+    
+    return 0;
+}
+
+int find_max(const string& infilename);
+int write_primes(const int size, bool primes[], const string& infilename, const string& outfilename);
+int write_composites(const int size, bool primes[], const string& infilename, const string& outfilename);
+void sieve(const int size, bool primes[]);
```

Comment:
no need to prototype the functions here, if they're defined (above) before they're used.  alternately, you could place these prototypes near the beginning of the file -- then the function definitions could be below `main()`, if you wanted.